plugins/imjournal/imjournal.c

```diff
@@ -415,13 +442,15 @@ readjournal(void)
 		tv.tv_usec = timestamp % 1000000;
 	}
 
+        /* This should become separate function, e.g. journal_context_update_cursor() for example */
 	/* save journal cursor (at this point we can be sure it is valid) */
 	sd_journal_get_cursor(j, &c);
 	if (c) {
 		free(last_cursor);
 		last_cursor = c;
 	}
 
+        /* Shouldn't we check error code and log on failure ? */
 	/* submit message */
 	enqMsg((uchar *)message, (uchar *) sys_iden_help, facility, severity, &tv, json, 0);
```

Comment:
Current idea is that if we reached this point we did not encounter any critical-enough error (each of the CHKiRet... skips below this on failure)